FRW/Services/Manager/WalletManager.swift

```diff
@@ -1077,6 +1088,18 @@ extension WalletManager {
     func fetchAccessible() async throws {
         try await accessibleManager.fetchFT()
     }
+    
+    func isEvmToken(_ token: TokenModel) -> Bool {
+        guard self.isSelectedEVMAccount else { return false }
+        guard let evmSupportedCoins else { return false }
+        return evmSupportedCoins.contains(where: { token.contractId == $0.contractId })
```

Comment:
Let's just check the token id only

Reply:
The best way to check this, is add inga enum in `TokenModel` which have .evm and .cadence.
Cause those token fetch from different endpoint, we know if they are EVM or Cadence at the beginning.

https://github.com/Outblock/FRW-iOS/blob/f3c15ff9689685dacc0a861612f1aaf70600d2f7/FRW/Foundation/Model/WalletModels.swift#L90-L101

Reply:
I could add a new field, but there are some issues:
- Decoding of existing items (such as cached) will inevitably fail unless a default value is provided
- To avoid the decoding failure, probably a manual implementation of encoding and decoding are required
- There is at least one case where the type is unknown when using `CustomToken`, which is then converted to `TokenModel`. 

I can't say if the type is relevant in the `CustomToken` case (if it is, then the new field should be optional), however I believe this is not a piece of information that belongs to the token itself, because it does not come from the source where the data is retrieved, but it is injected thereafter. A container should hold that information, which is actually what is currently done by using two separate collections.

However, If you believe this is needed, I can do it, but I need some clarifications to the three issues I mentioned above.

Reply:
Ok maybe I have found a way, I'll push a new commit shortly

Reply:
@lmcmz pls take a look at [this commit](https://github.com/Outblock/FRW-iOS/pull/683/commits/8be10b53cb9f5c9554b672dca7fe6c56bd75aac3) and let me know if it makes sense. If not, I'll revert.